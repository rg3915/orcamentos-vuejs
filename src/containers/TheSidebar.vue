<template>
  <CSidebar 
    fixed 
    :minimize="minimize"
    :show="show"
    @update:show="(value) => $store.commit('set', ['sidebarShow', value])"
  >
    <CSidebarBrand class="d-md-down-none" to="/">
      <CIcon 
        class="c-sidebar-brand-full" 
        name="logo" 
        size="custom-size" 
        :height="35" 
        viewBox="0 0 556 134"
      />
      <CIcon 
        class="c-sidebar-brand-minimized" 
        name="logo" 
        size="custom-size" 
        :height="35" 
        viewBox="0 0 110 134"
      />
    </CSidebarBrand>

    <CRenderFunction flat :content-to-render="$options.nav"/>
    <CSidebarMinimizer
      class="d-md-down-none"
      @click.native="$store.commit('set', ['sidebarMinimize', !minimize])"
    />
  </CSidebar>
</template>

<script>
import nav from './_nav'

export default {
  name: 'TheSidebar',
  nav,
  computed: {
    show () {
      return this.$store.state.sidebarShow 
    },
    minimize () {
      return this.$store.state.sidebarMinimize 
    }
  }
}
</script>
