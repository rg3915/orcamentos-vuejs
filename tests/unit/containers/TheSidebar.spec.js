import { shallowMount, createLocalVue } from '@vue/test-utils';
import CoreuiVue from '@coreui/vue'
import Vuex from 'vuex'
import TheSidebar from '@/containers/TheSidebar'

const localVue = createLocalVue()
localVue.use(CoreuiVue)
localVue.use(Vuex)

const store = new Vuex.Store({
  state: {
    sidebarShow: 'responsive',
    sidebarMinimize: false
  }
})

describe('TheSidebar.vue', () => {
  it('has a name', () => {
    expect(TheSidebar.name).toBe('TheSidebar')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(TheSidebar, { store, localVue })
    expect(wrapper.element).toMatchSnapshot()
  })
})
