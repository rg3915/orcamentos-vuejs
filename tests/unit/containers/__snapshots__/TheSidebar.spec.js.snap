// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSidebar.vue renders correctly 1`] = `
<csidebar-stub
  breakpoint="lg"
  colorscheme="dark"
  dropdownmode="openActive"
  fixed="true"
  hideonmobileclick="true"
  show="responsive"
>
  <csidebarbrand-stub
    activeclass="router-link-active"
    class="d-md-down-none"
    event="click"
    exactactiveclass="router-link-exact-active"
    routertag="a"
    tag="div"
    target="_self"
    to="/"
  >
    <cicon-stub
      class="c-sidebar-brand-full"
      height="35"
      name="logo"
      size="custom-size"
      viewbox="0 0 556 134"
    />
     
    <cicon-stub
      class="c-sidebar-brand-minimized"
      height="35"
      name="logo"
      size="custom-size"
      viewbox="0 0 110 134"
    />
  </csidebarbrand-stub>
   
  <crenderfunction-stub
    contenttorender="[object Object]"
    flat="true"
  />
   
  <csidebarminimizer-stub
    class="d-md-down-none"
  />
</csidebar-stub>
`;
